### PR TITLE
feat: add option to change the filename

### PR DESCRIPTION
**Pull request by ronen-e**
Optionally change file name

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ function CompressionPlugin(options) {
 	options = options || {};
 	this.asset = options.asset || "[path].gz[query]";
 	this.algorithm = options.algorithm || "gzip";
+	this.changeNewFileName = options.changeNewFileName || false;
```
Comment (michael-ciniawsky) [inline index.js]: `this.changeNewFileName` => `this.name` or something that is similiar terse 😛

Comment (bebraw) [inline index.js]: See how we did something similar in extract-text-webpack-plugin. Maybe a good solution to mimic.

Comment (michael-ciniawsky) [inline index.js]: @ronen-e please see https://github.com/webpack-contrib/extract-text-webpack-plugin#options (`options.filename`)

Comment (ronen-e) [inline index.js]: @michael-ciniawsky I didn't see any tests for the existing code.
I would like to add to them if they exist.
Do you have any tests that you already run on the code?

Comment (bebraw) [inline index.js]: Ah, yeah... We would have to update this project to **webpack-defaults** to get testing infrastructure in place.

Comment (michael-ciniawsky) [inline index.js]: @ronen-e Ooops..:stuck_out_tongue forget what I said then 🙃  Just change to`filename` + docs and good to go imho

**Pull request merged**